src/dao/db_context.rs

```diff
@@ -1,67 +1,79 @@
 use super::{Group, User};
+use sqlx::mysql::MySql;
 use sqlx::mysql::MySqlRow;
+use sqlx::pool::Pool;
 use sqlx::{FromRow, MySqlPool};
+use std::marker::PhantomData;
 use std::sync::Arc;
 
 pub struct Table<'c, T>
 where
-    T: FromRow<'c, MySqlRow<'c>>,
+    T: FromRow<'c, MySqlRow>,
 {
     pub pool: Arc<MySqlPool>,
-    _from_row: fn(&MySqlRow<'c>) -> Result<T, sqlx::Error>,
+    _from_row: Box<dyn Fn(&'c MySqlRow) -> Result<T, sqlx::Error> + Send + Sync + 'c>,
```

Comment:
Why does the callback `from_row` need to support `Send + Sync`? Is there something you're trying to support downstream? For example, I tried changing the `Table` definition to below and it builds and executes fine:
```rust
impl<'c, T> Table<'c, T>
where
    T: FromRow<'c, MySqlRow>,
{
    fn new(pool: Arc<MySqlPool>) -> Self {
        Table {
            pool,
            _from_row: T::from_row,
            _marker: PhantomData,
        }
    }
}
```
Same goes for `JoinTable`.

Reply:
I agree. I was thinking there might be scenarios where we could want to process rows in parallel or where the callback might be shared across threads, hence the Box and Send + Sync, but I agree that this brings unnecessary complexity for this context. I have updated the callback following your suggestion and pushed a new commit.